### PR TITLE
Skil 166 - Skill Builder Redesign 

### PR DESCRIPTION
**Pull request by aldiux4you**
Merged SKIL-170 branch to develop the Global CSS Style.  
Merged the SKIL-161-Confirm-Team branch to include components and the Build Team Page. 

- Moved the current logic of Navbar into AppState.js
- Created a Global Navbar that displays the logo and profile/settings.
- Created BasicTabs.js components to conditionally render Roster, Teams, and Assessment Task after navigating into a course. 
- Created Reusable Components (BackButton, BackButtonResource, CourseInfo, CustomDataTable, MainHeader).
- Add Global CSS Classes to SBStyles.css
- Refactored and redesigned logic for Add/Edit Course
- Moved logic of functions from AppState.js to its children components.

### CODE REVIEW

**Review by Wrimo — CHANGES_REQUESTED** [review]
First, I wanted to compliment how good the UI redesign looks. I poked around on the site a bit, and it looks very professional. 

Second, I don't really have much to say about the code. It generally all looks really good to my very limited knowledge of how to write React. The one thing I noticed that will need to be changed is that the new React code is not using the updates I made in the Code Refactor branch to avoid passing functions around and instead pass a reference to the navbar. All of that will need to be changed before merging into master.

**Review by Wrimo — CHANGES_REQUESTED** [review]

File: BackEndFlask/controller/Routes/User_routes.py [hunk 1]
```diff
@@ -60,9 +60,10 @@ def getAllUsers():
                         createBadResponse(f"An error occurred retrieving all users enrolled in course_id: {course_id}!", admin_user, "users")
                         return response
                     all_users.append(admin_user)
-                for role in request.args.getlist("role_id"):
-                    if user.role_id is int(role):
-                        all_users.append(user)
+                else:
```
[the remark this answers is not in the document]
Comment (brianlugo1) [reply BackEndFlask/controller/Routes/User_routes.py:63]: The following code is flawed and has been refactored in SKIL-174. Once we merge this pull request into master before merging SKIL-174, the issue will be resolved!

**Review by Wrimo — APPROVED** [review]
LGTM

**Review by brianlugo1 — APPROVED** [review]
@sah0017 LGTM!

**Pull request merged**